### PR TITLE
feat: implemented usage of exisitng remote ID

### LINKED ISSUE #725: Add Known Remote Peer ID to Install Steps
**Is your feature request related to a problem? Please describe.**
Currently when a user installs the PWA and wants to connect to a remote Ambianic Edge Device the install sequence forces the user to send a message to another user who has a remote peer ID to connect to. See screenshot below. 

As it turns out in some cases the user already has the Peer ID and does not need to send a request message. They already obtained the ID outside the PWA app.

In order to avoid being stuck at this step, the user has to Skip the whole install sequence and go to the Settings page where they can enter the known remote peer ID. While this works, its not the most intuitive experience for a new PWA install.

**Describe the solution you'd like**
It would be desirable to have an option for the user to choose whether they already have the remote edge device peer ID and just enter it without having to send a request message.
Screen Shot 2021-09-08 at 12 30 18 PM

### PR DESCRIPTION
**Pull request by vickywane**
## Purpose
This pull request will close [this issue](https://github.com/ambianic/ambianic-ui/issues/725).

## Approach
Work done within this pull request modifies the installation steps within the onboarding page to allow a user enter the remote peer ID of an edge device when onboarding an existing edge device locally.

The screenshots below shows the changes made to the onboarding flow. 
- The added text for a user having an existing remote peer ID.
Screenshot_20210922_100833

- Text input field for a user to type the remote peer for an edge device. 
Screenshot_20210922_100921

The MP4 video below shows the entire added flow;
 
https://user-images.githubusercontent.com/29664439/134736277-956ac1a0-280e-4810-ad57-797a2fc5ad2f.mp4

## Merge Checklist

- [x]  The code change is tested and works locally.
- [x]  The user and dev documentation is up to date.
- [x]  There is no commented out code in this PR.
- [x]  No lint errors (use flake8)
- [x]  100% test coverage

### CODE REVIEW

**Comment by vickywane** [comment]
This pull request is fully ready for review. 

The netlify checks failing is due to the absence of a `yarn.lock` file. I removed that since we are sticking to NPM.

**Review by ivelin — CHANGES_REQUESTED** [review]
There are 6 failing checks. Please go through each and fix.

**Comment by vickywane** [comment]
@ivelin This pull request is now ready for  review as I have taken care of the DCO check, and added a video to show a locally tested version of the new flow.

File: cypress/integration/ambianic-tests/onboarding.spec.js [hunk 1]
```diff
@@ -23,19 +25,29 @@ context('Onboarding Wizard', () => {
     cy.get('[data-cy=continue-to-edge-installation]').click()
   })
 
-  it('It confirms remote request message and request grant', () => {
```
Comment (ivelin) [inline cypress/integration/ambianic-tests/onboarding.spec.js:26]: @vickywane why did you get rid of the request grant test? It's still a valid scenario we need to cover. Check that the request dialogs show up, etc. Test that the offered request mediums show up as expected: email, message.

File: src/views/Onboarding.vue [hunk 2]
```diff
@@ -199,7 +199,7 @@
                   </v-btn>
                 </div>
 
-                <div v-if="stepContentName == 'edge-installation-question'">
+                <div v-if="stepContentName === 'edge-installation-question'">
```
Comment (ivelin) [inline src/views/Onboarding.vue:202]: @vickywane this file has become too large. Nearly 1,000 lines. It's nearly impossible to follow the logic and the interdependency between states. I spent a lot of time resolving issues with the Install PWA step, which was broken and left the use with the impression that the app was installed, when in reality it was just eating silently browser exceptions. The code in this one file became too convoluted. Please break it up into ~200 LOC files that can be independently tested.

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane good flow. Please break up the large convoluted file into maintainable modules and increase test coverage.

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane Nice design! Please see comments on adding unit tests, namespacing and some cleanup.

**Comment by vickywane** [comment]
> Entering direct remote peer ID gets stuck. The error message in the log that "PeerJS: Error ID... is taken" indicates that we are causing competing instances of PeerJS to establish remote connection. Need to debug what causes this. We should make sure that the app uses exactly one active peerjs instance.

I think this issue is a known problem similar to [bug report](https://github.com/ambianic/ambianic-ui/issues/710) and is being worked upon within this [pull request](https://github.com/ambianic/ambianic-ui/pull/727).

**Comment by vickywane** [comment]
> 2. When a remote peer ID is entered, the UI shows some activity (see dev console logs) but does not result in the expected connection. After click on Cancel, and then submitting the remote Peer ID again, then the connection happens as expected.

This works normally when i try to reproduce this.

Can you please show a recording for this? 
Also, could this be a case of the peer-network fluctuating?

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane looking good. Almost there.

Please check the two unresolved conversation
Screen Shot 2021-10-10 at 1 21 32 PM
s.

**Comment by vickywane** [comment]
> Please check the two unresolved conversation

I have responded to the comment on cypress test. 

I'm working on the Vuex namespace comment.

**Comment by vickywane** [comment]
@ivelin This [GitHub issue](https://github.com/ambianic/ambianic-ui/issues/755) has been created to track the work done to introduce namespaces within the PWA.

**Review by ivelin — CHANGES_REQUESTED** [review]
@vickywane see comments related to use of local storage.

**Review by ivelin — CHANGES_REQUESTED** [review]
Still not ready. Please Do Not submit another hasty PR. Test thoroughly manually and with unit tests before requesting review again.

**Review by ivelin — CHANGES_REQUESTED** [review]
Almost there. 
The label "Already have a Remote Peer ID" is barely visible on my screen. See screenshot. Let's use a default color instead of a custom color that is not tested on all typical user screens.

See the other comment about still inconsistent variable naming.

Screen Shot 2021-10-19 at 9 03 46 AM

**Review by ivelin — CHANGES_REQUESTED** [review]
address open comments

**Review by ivelin — CHANGES_REQUESTED** [review]
"Already have a valid Peer ID" label is still barely visible. See screenshot

Screen Shot 2021-10-19 at 3 05 19 PM
.

**Review by ivelin — CHANGES_REQUESTED** [review]
First of all, CI checks are failing.
Second, I asked for two other reviewers to test and approve the UX thoroughly before submitting back to me.
Thank you.

**Comment by ivelin** [comment]
This is now addressed via #757

**Pull request closed without merging**